.github/workflows/ci.yml

```diff
@@ -0,0 +1,52 @@
+name: CI
+on:
+  - push
+  - pull_request
+jobs:
+  test:
+    name: Julia ${{ matrix.version }} - ${{ matrix.os }} - ${{ matrix.arch }} - ${{ github.event_name }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - version: 'nightly'
+            os: ubuntu-latest
+            arch: x64
+          - version: '1'
+            os: ubuntu-latest
+            arch: x64
+          - version: '1.0'
+            os: ubuntu-latest
+            arch: x64
+          - version: '1.0'
+            os: ubuntu-latest
+            arch: x86
+          - version: '1'
+            os: windows-latest
+            arch: x64
+          - version: '1'
+            os: macOS-latest
+            arch: x64
```

Comment:
The rule-of-thumb I went with was:
- Does the package have binary dependencies?
  - If yes, test on Linux, Mac, and Window 64-bit, Julia 1 and Julia 1.0
    - Also test on linux 32-bit Julia 1.0 for `Int32` mistakes
    - https://github.com/jump-dev/GLPK.jl/blob/master/.github/workflows/ci.yml
  - If no, test on Linux 64-bit, Julia 1 and 1.0, and Linux 32-bit, Julia 1.0
    - Optionally test on Window 64-bit, Julia 1.0 to test windows installation
    - https://github.com/jump-dev/MathOptInterface.jl/blob/master/.github/workflows/ci.yml

Reply:
recommendation followed in last commit